log/adapter_test.go

```diff
@@ -0,0 +1,159 @@
+			want: `DEBUG {"int":1,"int64":9223372036854775807,"string":string,"bool":true,"time.Duration":1h0m0s,"[]string":[Abc Def Ghi],"named error":named error,"error":error,"int":1,"int64":9223372036854775807,"string":any string,"bool":true,"[]string":[Abc Def Ghi],"error":error,"struct":{test},"stringer":stringerTest}`,
+		},
+		{
+			name: "logs.ERROR",
+			a:    logger,
+			args: args{opts: logs.Options{Lvl: logs.ERROR}, msg: "ERROR", fields: fields},
+			want: `ERROR {"int":1,"int64":9223372036854775807,"string":string,"bool":true,"time.Duration":1h0m0s,"[]string":[Abc Def Ghi],"named error":named error,"error":error,"int":1,"int64":9223372036854775807,"string":any string,"bool":true,"[]string":[Abc Def Ghi],"error":error,"struct":{test},"stringer":stringerTest}`,
+		},
+		{
+			name: "logs.FATAL",
+			a:    logger,
+			args: args{opts: logs.Options{Lvl: logs.FATAL}, msg: "FATAL", fields: fields},
+			want: `FATAL {"int":1,"int64":9223372036854775807,"string":string,"bool":true,"time.Duration":1h0m0s,"[]string":[Abc Def Ghi],"named error":named error,"error":error,"int":1,"int64":9223372036854775807,"string":any string,"bool":true,"[]string":[Abc Def Ghi],"error":error,"struct":{test},"stringer":stringerTest}`,
+		},
+		{
+			name: "logs.INFO",
+			a:    logger,
+			args: args{opts: logs.Options{Lvl: logs.INFO}, msg: "INFO", fields: fields},
+			want: `INFO {"int":1,"int64":9223372036854775807,"string":string,"bool":true,"time.Duration":1h0m0s,"[]string":[Abc Def Ghi],"named error":named error,"error":error,"int":1,"int64":9223372036854775807,"string":any string,"bool":true,"[]string":[Abc Def Ghi],"error":error,"struct":{test},"stringer":stringerTest}`,
+		},
+		{
+			name: "logs.QUIET",
+			a:    logger,
+			args: args{opts: logs.Options{Lvl: logs.QUIET}, msg: "QUIET", fields: fields},
+			want: ``,
+		},
+		{
+			name: "logs.TRACE",
+			a:    logger,
+			args: args{opts: logs.Options{Lvl: logs.TRACE}, msg: "TRACE", fields: fields},
+			want: `TRACE {"int":1,"int64":9223372036854775807,"string":string,"bool":true,"time.Duration":1h0m0s,"[]string":[Abc Def Ghi],"named error":named error,"error":error,"int":1,"int64":9223372036854775807,"string":any string,"bool":true,"[]string":[Abc Def Ghi],"error":error,"struct":{test},"stringer":stringerTest}`,
+		},
+		{
+			name: "logs.WARN",
+			a:    logger,
+			args: args{opts: logs.Options{Lvl: logs.WARN}, msg: "WARN", fields: fields},
+			want: `WARN {"int":1,"int64":9223372036854775807,"string":string,"bool":true,"time.Duration":1h0m0s,"[]string":[Abc Def Ghi],"named error":named error,"error":error,"int":1,"int64":9223372036854775807,"string":any string,"bool":true,"[]string":[Abc Def Ghi],"error":error,"struct":{test},"stringer":stringerTest}`,
+		},
+		{
+			name: "Unknown level",
+			a:    logger,
+			args: args{opts: logs.Options{Lvl: logs.Level(99)}, msg: "Unknown level", fields: fields},
+			want: `unknown level`,
+			fail: true,
+		},
+	}
+	for _, tt := range tests {
```

Comment:
There is no check about all types was checked.

What about make ```types  := map[FieldType]bool{}``` and for every field mark ```types[field. ftype] = true```, then check about all types from 0 to EndType was present?